### PR TITLE
[ARTEMIS-2130]Web console display blank ClientID for the core client

### PR DESCRIPTION
**Pull request by shailendra14k**
JIRA:- https://issues.apache.org/jira/browse/ARTEMIS-2130

While using core client. Connection.setClientID(String) , set the ServerSession metadata using the key "jms-client-id"

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java [hunk 1]
```diff
@@ -51,10 +51,14 @@ public JsonObjectBuilder toJson(RemotingConnection connection) {
 
       List<ServerSession> sessions = server.getSessions(connection.getID().toString());
       Set<String> users = new HashSet<>();
+      String clientID = null;
 
       for (ServerSession session : sessions) {
          String username = session.getUsername() == null ? "" : session.getUsername();
          users.add(username);
+         if (clientID == null) {
+            clientID = session.getMetaData("jms-client-id") == null ? connection.getClientID() : session.getMetaData("jms-client-id");
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java]: should avoid jms specifics such as this.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java [hunk 2]
```diff
@@ -91,7 +95,14 @@ public Object getField(RemotingConnection connection, String fieldName) {
          case "protocol":
             return connection.getProtocolName();
          case "clientID":
-            return connection.getClientID();
+            String clientID = null;
+            for (ServerSession session : sessions) {
+               if (clientID == null) {
+                  clientID = session.getMetaData("jms-client-id") == null ? connection.getClientID() : session.getMetaData("jms-client-id");
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java]: jms specifics should not be in core broker. Also this is the connection view, this data is off the session view, so should be visible there, not on the connection.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConsumerView.java [hunk 3]
```diff
@@ -82,7 +82,7 @@ public Object getField(ServerConsumer consumer, String fieldName) {
          case "user":
             return session.getUsername();
          case "clientID":
-            return consumer.getConnectionClientID();
+            return session.getMetaData("jms-client-id") == null ? consumer.getConnectionClientID() : session.getMetaData("jms-client-id");
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConsumerView.java]: this should not be needed, if anything the session and consumer models should be re-worked to allow for a client id to be set at session level generically.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ProducerView.java [hunk 4]
```diff
@@ -79,7 +79,7 @@ public Object getField(ServerProducer producer, String fieldName) {
          case "user":
             return session.getUsername();
          case "clientID":
-            return session.getRemotingConnection().getClientID();
+            return session.getMetaData("jms-client-id") == null ? session.getRemotingConnection().getClientID() : session.getMetaData("jms-client-id");
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ProducerView.java]: this should not be needed, if anything the session and producer models should be re-worked to allow for a client id to be set at session level generically.

**Comment by shailendra14k** [comment]
@michaelandrepearce @jbertram Thank you for the review. Yes, session and consumer models need to be re-worked and should have a generic method to get/set client_id accros the different client. 

I am closing this PR.

**Pull request closed without merging**